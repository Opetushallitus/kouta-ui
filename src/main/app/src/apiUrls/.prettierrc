{
  "printWidth": 1000,
  "trailingComma": "all",
  "tabWidth": 2,
  "semi": true,
  "singleQuote": true
}