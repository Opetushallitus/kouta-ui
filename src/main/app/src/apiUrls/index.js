import { development as developmentUrls } from './urls';

export const configure = async urls => {
  const { NODE_ENV, REACT_APP_CYPRESS } = process.env;

  const isCypress = !!REACT_APP_CYPRESS;

  if (['development', 'test'].includes(NODE_ENV)) {
    urls.addProperties(developmentUrls({ isCypress }));
  } else {
    await urls.load({ overrides: '/kouta/rest/config/frontProperties' });
  }

  return urls;
};

export default configure;
