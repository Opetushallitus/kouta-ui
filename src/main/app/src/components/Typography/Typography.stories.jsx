import React from 'react';
import { storiesOf } from '@storybook/react';

import Typography from './index';

storiesOf('Typography', module)
  .add('Basic', () => <Typography>Hello world!</Typography>)
  .add('With variant', () => (
    <>
      <Typography variant="body">Body (default)</Typography>
      <Typography variant="h1">Heading 1</Typography>
      <Typography variant="h2">Heading 2</Typography>
      <Typography variant="h3">Heading 3</Typography>
      <Typography variant="h4">Heading 4</Typography>
      <Typography variant="h5">Heading 5</Typography>
      <Typography variant="h6">Heading 6</Typography>
    </>
  ))
  .add('With spacing', () => (
    <>
      <Typography
        variant="h1"
        marginTop={1}
        marginBottom={2}
        marginLeft={3}
        marginRight={4}
      >
        Margin
      </Typography>
      <Typography
        variant="h1"
        paddingTop={1}
        paddingBottom={2}
        paddingLeft={3}
        paddingRight={4}
      >
        Padding
      </Typography>
    </>
  ));
