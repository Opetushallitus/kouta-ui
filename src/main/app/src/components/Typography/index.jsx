export { default } from '@opetushallitus/virkailija-ui-components/Typography';
