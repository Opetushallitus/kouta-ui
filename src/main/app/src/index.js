import 'core-js/features/object/keys';
import 'core-js/features/object/values';
import 'core-js/features/promise';
import 'core-js/features/array/includes';
import 'core-js/features/array/every';
import 'core-js/features/array/flat-map';

import React from 'react';
import ReactDOM from 'react-dom';
import createBrowserHistory from 'history/createBrowserHistory';
import { urls as ophUrls } from 'oph-urls-js';
import get from 'lodash/get';
import merge from 'lodash/merge';

import * as serviceWorker from './serviceWorker';
import App from './components/App';
import createStore from './state';
import defaultTheme from './theme';
import configureUrls from './apiUrls';
import createLocalisation from './localisation';
import { getLocalisation } from './apiUtils';
import getTranslations from './translations';
import createHttpClient from './httpClient';

const history = createBrowserHistory({ basename: 'kouta' });

serviceWorker.unregister();

const isDev = process.env.NODE_ENV === 'development';

const loadLocalisation = async ({
  namespace,
  language,
  httpClient,
  apiUrls,
}) => {
  const localisation = await getLocalisation({
    category: namespace,
    locale: language,
    httpClient,
    apiUrls,
  });

  const translations = getTranslations();

  return get(translations, [language, namespace])
    ? merge({}, translations[language][namespace], localisation || {})
    : localisation;
};

(async () => {
  const apiUrls = await configureUrls(ophUrls);

  const httpClient = createHttpClient({
    apiUrls,
    callerId: !isDev ? process.env.REACT_APP_CALLER_ID : null,
  });

  const localisationInstance = createLocalisation({
    debug: isDev,
    loadLocalisation: ({ namespace, language }) =>
      loadLocalisation({ namespace, language, httpClient, apiUrls }),
  });

  window.__i18n__ = localisationInstance;

  const { store, persistor } = createStore({
    apiUrls,
    httpClient,
    history,
    localisation: localisationInstance,
  });

  window.__store__ = store;

  ReactDOM.render(
    <App
      store={store}
      theme={defaultTheme}
      urls={apiUrls}
      httpClient={httpClient}
      history={history}
      localisation={localisationInstance}
      persistor={persistor}
    />,
    document.getElementById('root'),
  );
})();
