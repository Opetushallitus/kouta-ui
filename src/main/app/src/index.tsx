import React from 'react';

import { Globals } from '@react-spring/web';
import { inspect } from '@xstate/inspect';
import { createBrowserHistory } from 'history';
import { urls as ophUrls } from 'oph-urls-js';
import ReactDOM from 'react-dom';

import createHttpClient from './httpClient';
import { createDefaultLocalization } from './localization';
import App from './pages/App';
import { createStore } from './state';
import defaultTheme from './theme';
import { configure as configureUrls } from './urls';
import { isPlaywright } from './utils';

Globals.assign({
  skipAnimation: isPlaywright,
});

if (import.meta.env.VITE_XSTATE_INSPECTOR) {
  inspect({
    iframe: false, // open in new window
  });
}

const history = createBrowserHistory({ basename: 'kouta' });

(async () => {
  let apiUrls = ophUrls;

  const httpClient = createHttpClient({
    apiUrls,
    callerId: import.meta.env.VITE_CALLER_ID,
  });

  apiUrls = await configureUrls(apiUrls, httpClient);

  const localizationInstance = await createDefaultLocalization({
    httpClient,
    apiUrls,
  });

  const { store, persistor } = createStore();

  ReactDOM.render(
    <App
      store={store}
      theme={defaultTheme}
      urls={apiUrls}
      httpClient={httpClient}
      history={history}
      localization={localizationInstance}
      persistor={persistor}
    />,
    document.getElementById('root')
  );
})();
