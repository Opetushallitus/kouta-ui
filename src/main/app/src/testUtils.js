import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const shallowWithTheme = makeFnWithTheme(shallow);

export const mountWithTheme = makeFnWithTheme(mount);

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => +date;

  return () => {
    global.Date = _Date;
  };
};
