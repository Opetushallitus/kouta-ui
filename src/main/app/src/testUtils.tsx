import React, { PropsWithChildren } from 'react';

import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import { createRootReducer } from './state/rootReducer';
import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const renderWithTheme = makeFnWithTheme(render);

export const mountWithTheme = tree => {
  const { container } = makeFnWithTheme(render)(tree);
  return container.firstChild;
};

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => Number(date);

  return () => {
    global.Date = _Date;
  };
};

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: createRootReducer({}), preloadedState }),
    ...renderOptions
  }
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
