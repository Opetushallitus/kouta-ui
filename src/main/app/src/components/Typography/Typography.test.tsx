import React from 'react';

import Typography from './index';
import { mountWithTheme } from '#/src/testUtils';

test('renders correctly', () => {
  expect(
    mountWithTheme(<Typography>Hello world!</Typography>)
  ).toMatchSnapshot();
});

test('renders correctly with variant', () => {
  expect(
    mountWithTheme(
      <>
        <Typography variant="body">Body</Typography>
        <Typography variant="h1">Heading 1</Typography>
        <Typography variant="h2">Heading 2</Typography>
        <Typography variant="h3">Heading 3</Typography>
        <Typography variant="h4">Heading 4</Typography>
        <Typography variant="h5">Heading 5</Typography>
        <Typography variant="h6">Heading 6</Typography>
      </>
    )
  ).toMatchSnapshot();
});
