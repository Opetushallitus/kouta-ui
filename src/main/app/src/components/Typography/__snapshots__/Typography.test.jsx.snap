// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  color: #666666;
  line-height: 1.5;
  margin-bottom: 0px;
  margin-top: 0px;
  font-size: 1rem;
  color: #666666;
  line-height: 1.5;
  font-family: 'Roboto',sans-serif;
}

<styled.span
  as="span"
  variant="body"
>
  <span
    className="c0"
  >
    Hello world!
  </span>
</styled.span>
`;

exports[`renders correctly with variant 1`] = `
Array [
  .c0 {
  font-family: 'Roboto',sans-serif;
  color: #666666;
  line-height: 1.5;
  margin-bottom: 0px;
  margin-top: 0px;
  font-size: 1rem;
  color: #666666;
  line-height: 1.5;
  font-family: 'Roboto',sans-serif;
}

<styled.span
    as="span"
    variant="body"
  >
    <span
      className="c0"
    >
      Body
    </span>
  </styled.span>,
  .c0 {
  font-family: 'Roboto',sans-serif;
  color: #666666;
  line-height: 1.5;
  margin-bottom: 0px;
  margin-top: 0px;
  color: #2a2a2a;
  font-family: 'Roboto',sans-serif;
  font-size: 3rem;
  font-weight: 500;
  line-height: 1.2;
}

<styled.span
    as="h1"
    variant="h1"
  >
    <h1
      className="c0"
    >
      Heading 1
    </h1>
  </styled.span>,
  .c0 {
  font-family: 'Roboto',sans-serif;
  color: #666666;
  line-height: 1.5;
  margin-bottom: 0px;
  margin-top: 0px;
  color: #2a2a2a;
  font-family: 'Roboto',sans-serif;
  font-size: 2.5rem;
  font-weight: 500;
  line-height: 1.2;
}

<styled.span
    as="h2"
    variant="h2"
  >
    <h2
      className="c0"
    >
      Heading 2
    </h2>
  </styled.span>,
  .c0 {
  font-family: 'Roboto',sans-serif;
  color: #666666;
  line-height: 1.5;
  margin-bottom: 0px;
  margin-top: 0px;
  color: #2a2a2a;
  font-family: 'Roboto',sans-serif;
  font-size: 2rem;
  font-weight: 500;
  line-height: 1.2;
}

<styled.span
    as="h3"
    variant="h3"
  >
    <h3
      className="c0"
    >
      Heading 3
    </h3>
  </styled.span>,
  .c0 {
  font-family: 'Roboto',sans-serif;
  color: #666666;
  line-height: 1.5;
  margin-bottom: 0px;
  margin-top: 0px;
  color: #2a2a2a;
  font-family: 'Roboto',sans-serif;
  font-size: 1.5rem;
  font-weight: 500;
  line-height: 1.2;
}

<styled.span
    as="h4"
    variant="h4"
  >
    <h4
      className="c0"
    >
      Heading 4
    </h4>
  </styled.span>,
  .c0 {
  font-family: 'Roboto',sans-serif;
  color: #666666;
  line-height: 1.5;
  margin-bottom: 0px;
  margin-top: 0px;
  color: #2a2a2a;
  font-family: 'Roboto',sans-serif;
  font-size: 1.25rem;
  font-weight: 500;
  line-height: 1.2;
}

<styled.span
    as="h5"
    variant="h5"
  >
    <h5
      className="c0"
    >
      Heading 5
    </h5>
  </styled.span>,
  .c0 {
  font-family: 'Roboto',sans-serif;
  color: #666666;
  line-height: 1.5;
  margin-bottom: 0px;
  margin-top: 0px;
  color: #2a2a2a;
  font-family: 'Roboto',sans-serif;
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.2;
}

<styled.span
    as="h6"
    variant="h6"
  >
    <h6
      className="c0"
    >
      Heading 6
    </h6>
  </styled.span>,
]
`;
